### PR TITLE
feat: support PromQL operations over the same metric

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This patch fixes a problem that when join on the same table(metric), operations will lead to wrong result due to qualifier ambiguity. E.g.: `(sum by(host) (irate(host_cpu_seconds_total{mode!="idle"}[1m0s])) / sum by (host)((irate(host_cpu_seconds_total[1m0s])))) * 100;` which calculate the CPU utilies. Before this change it will always return `1`

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/promql/src/planner.rs [hunk 1]
```diff
@@ -1847,11 +1853,13 @@ impl PromPlanner {
             .zip(self.ctx.field_columns.iter())
             .map(|(expr, name)| Ok(DfExpr::Alias(Alias::new(expr, name.to_string()))));
 
-        // chain non-value columns (unchanged) and value columns (applied computation then alias)
+        // chain non-field columns (unchanged) and field columns (applied computation then alias)
         let project_fields = non_field_columns_iter
             .chain(field_columns_iter)
             .collect::<Result<Vec<_>>>()?;
 
+        common_telemetry::info!("[DEBUG] project fields: {:?}", project_fields);
```
Comment (killme2008) [inline src/promql/src/planner.rs:1861]: Remove the log or change it to debug level

**Review by killme2008 — APPROVED** [review]
Almostly LGTM

**Comment by killme2008** [comment]
@yuanbohan PTAL

File: src/promql/src/planner.rs (hunk 1 above)
Comment (waynexia) [reply src/promql/src/planner.rs:1861]: ah, I forget this

**Review by yuanbohan — APPROVED** [review]

**Pull request merged**